core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java

```diff
@@ -191,14 +191,21 @@ public IOContextResult handleIO(NetworkIOJob netIoJob) {
     }
 
     @Override
-    public LineTcpConnectionContext of(int fd, IODispatcher<LineTcpConnectionContext> dispatcher) {
-        authenticator.init(fd, recvBufStart, recvBufEnd, 0, 0);
+    public LineTcpConnectionContext of(Socket socket, IODispatcher<LineTcpConnectionContext> dispatcher) {
+        if (socket != null) {
+            if (socket.supportsTls()) {
```

Comment:
how do we end up with `null` socket here? If this scenario exists, it should be documented in comments

Reply:
`IOContextFactoryImpl` needs to clean up the context for potential reuse. See https://github.com/questdb/questdb/blob/c9c85f114ab5a3bbedd9656031b5fc025893db61/core/src/main/java/io/questdb/network/IOContextFactoryImpl.java#L55

I'll add `@Nullable` annotations to the arguments to make it more apparent.

Reply:
Added in 3d8e9c6668c029b8c86be4b37f3c686db8cbcbe4